tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/llk_math_eltwise_unary_sfpu_macros.h

```diff
@@ -0,0 +1,176 @@
+
+// For ops without extra uint parameter
+#define SFPU_UNARY_NO_PARAM_KERNEL(OP, MODE, APPROXIMATE, DST_IDX) \
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(             \
+        ckernel::sfpu::calculate_##OP<APPROXIMATE>, DST_IDX, (int)VectorMode::MODE)
+
+// For ops without extra uint parameter with type
+#define SFPU_UNARY_NO_PARAM_KERNEL_WITH_TYPE(OP, TYPE, MODE, APPROXIMATE, DST_IDX) \
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(                             \
+        ckernel::sfpu::calculate_##OP<SfpuType::TYPE, APPROXIMATE>, DST_IDX, (int)VectorMode::MODE)
+
+// For ops where compute takes extra args
+#define SFPU_UNARY_PARAMS_KERNEL_EXTRA_ARGS(FN, MODE, APPROXIMATE, DST_IDX, PARAM0, PARAM1) \
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(                                      \
+        ckernel::sfpu::FN<APPROXIMATE>, DST_IDX, (int)VectorMode::MODE, PARAM0, PARAM1);
+
+// For ops with multiple template parameters and one runtime parameter (e.g., scale)
+#define SFPU_TEMPLATE_PARAMS_KERNEL(                                                                                  \
+    OP,                                                                                                               \
+    APPROXIMATE,                                                                                                      \
+    FAST_APPROX,                                                                                                      \
+    SCALE_EN,                                                                                                         \
+    SKIP_POSITIVE_CHECK,                                                                                              \
+    ITERATIONS,                                                                                                       \
+    IS_FP32_DEST_ACC_EN,                                                                                              \
+    DST_IDX,                                                                                                          \
+    VECTOR_MODE,                                                                                                      \
+    SCALE)                                                                                                            \
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(                                                                \
+        ckernel::sfpu::                                                                                               \
+            calculate_##OP<APPROXIMATE, FAST_APPROX, SCALE_EN, ITERATIONS, SKIP_POSITIVE_CHECK, IS_FP32_DEST_ACC_EN>, \
+        DST_IDX,                                                                                                      \
+        VECTOR_MODE,                                                                                                  \
+        SCALE)
```

Comment:
Could you elaborate on this particular order of parameters chosen? Was just curious how you determined this ordering since it doesn't completely align with the order of template parameters in the expansion.

I'd also personally prefer to make `calculate_` part of the first parameter rather than concatenating it, since that will help improve code navigation which you explicitly stated as one of your goals.